public/app/core/components/ErrorBoundary/ErrorBoundary.tsx

```diff
@@ -0,0 +1,47 @@
+﻿import React, { Component } from 'react';
+
+interface ErrorInfo {
+  componentStack: string;
+}
+
+interface RenderProps {
+  error: Error;
+  errorInfo: ErrorInfo;
+}
+
+interface Props {
+  children: (r: RenderProps) => JSX.Element;
+}
+
+interface State {
+  error: Error;
+  errorInfo: ErrorInfo;
+}
+
+class ErrorBoundary extends Component<Props, State> {
+  constructor(props) {
+    super(props);
+    this.state = { error: null, errorInfo: null };
```

Comment:
As the state does not rely on props, we could move it to the class level and get rid of the constructor:

```ts
readonly state: State = {
  error: null;
  errorInfo: null
}
```

Reply:
Nice!